lib/arke/exchange/bitfinex.rb

```diff
@@ -22,14 +24,43 @@ def process_channel_message(msg)
       data = msg[1]
 
       if data.length == 3
-        update_order(data)
+        process_data(data)
       elsif data.length > 3
-        data.each { |order| update_order(order) }
+        data.each { |order| process_data(order) }
       end
     end
 
-    def update_order(data)
-      Arke::Log.info "Got Order: #{data}"
+    def process_data(data)
+      order = parse_ws_data(data)
+      order.amount.zero? ? push(:cancel_order, order) : push(:create_order, order)
+    end
+
+    def parse_ws_data(data)
+      data[2].positive? ? build_buy_order(data) : build_sell_order(data)
+    end
+
+    def push(action, order)
+      @strategy.push(Arke::Action.new(action, order))
+    end
+
```

Comment:
todo: avoid duplicating code, also order can be smarter and use the same convention

Reply:
I guess we better move it to Base class